common/cpp/react/renderer/components/safeareacontext/RNCSafeAreaViewShadowNode.cpp

```diff
@@ -0,0 +1,107 @@
+#include "RNCSafeAreaViewShadowNode.h"
+
+#include <react/renderer/components/view/conversions.h>
+#include <react/renderer/core/LayoutContext.h>
+#include <yoga/Yoga.h>
+#include <algorithm>
+
+namespace facebook {
+namespace react {
+
+extern const char RNCSafeAreaViewComponentName[] = "RNCSafeAreaView";
+
+inline YGValue valueFromEdges(YGStyle::Edges edges, YGEdge edge, YGEdge axis) {
+  YGValue edgeValue = edges[edge];
+  if (edgeValue.unit != YGUnitUndefined) {
+    return edgeValue;
+  }
+  YGValue axisValue = edges[axis];
+  if (axisValue.unit != YGUnitUndefined) {
+    return axisValue;
+  }
+  return edges[YGEdgeAll];
+}
+
+void RNCSafeAreaViewShadowNode::adjustLayoutWithState() {
+  ensureUnsealed();
+
+  auto props = getConcreteProps();
+  auto state =
+      std::static_pointer_cast<const RNCSafeAreaViewShadowNode::ConcreteState>(
+          getState());
+  auto stateData = state->getData();
+  auto edges = props.edges;
+
+  // Get the current values for padding / margin. The only caveat here is that
+  // percent values are not supported. Also might need to add support for start
+  // / end.
+  YGValue top, left, right, bottom;
+  if (props.mode == RNCSafeAreaViewMode::Padding) {
+    top = valueFromEdges(props.yogaStyle.padding(), YGEdgeTop, YGEdgeVertical);
+    left =
+        valueFromEdges(props.yogaStyle.padding(), YGEdgeLeft, YGEdgeHorizontal);
+    bottom =
+        valueFromEdges(props.yogaStyle.padding(), YGEdgeBottom, YGEdgeVertical);
+    right = valueFromEdges(
+        props.yogaStyle.padding(), YGEdgeRight, YGEdgeHorizontal);
+  } else {
+    top = valueFromEdges(props.yogaStyle.margin(), YGEdgeTop, YGEdgeVertical);
+    left =
+        valueFromEdges(props.yogaStyle.margin(), YGEdgeLeft, YGEdgeHorizontal);
+    bottom =
+        valueFromEdges(props.yogaStyle.margin(), YGEdgeBottom, YGEdgeVertical);
+    right =
+        valueFromEdges(props.yogaStyle.margin(), YGEdgeRight, YGEdgeHorizontal);
+  }
+  if (std::find(edges.begin(), edges.end(), "top") != edges.end()) {
```

Comment:
I think you could do following:

```
edges.find("top") != edges.end()
```

same result but it is easier to read.

I think free function `std::find` is meant to be used in templates.

Reply:
I don't think find exists for std::vector, am I missing something?

Ideally this would be a bitmask or set, but that's what is generated by codegen so I can't change it.